src/lib/Libutil/misc_utils.c

```diff
@@ -2568,20 +2563,34 @@ get_range_from_jid(char *jid)
 	return index;
 }
 
+/**
+ * @brief
+ * 	create and return (in a static array) a jobid for a subjob based on
+ * 	the parent jobid and the subjob index
+ *
+ * @param[in] parent_jid - parent jobid
+ * @param[in] sjidx -  subjob index.
+ *
+ * @return char *
+ * @return jobid of subjob
+ *
+ * @par
+ * 	MT-safe: No - uses a static buffer, "jid".
+ */
 char *
 create_subjob_id(char *parent_jid, int sjidx)
 {
-	static char jid[PBS_MAXSVRJOBID+1];
-	char        index[20];
-	char       *pb;
+	static char jid[PBS_MAXSVRJOBID + 1];
+	char index[20];
+	char *pb;
 
 	sprintf(index, "%d", sjidx);
-	strcpy(jid, parent_jid);
+	pbs_strncpy(jid, parent_jid, sizeof(jid));
 
-	pb = strchr(jid, (int)'[');	/* "seqnum[" section */
-	*(pb+1) = '\0';
-	strcat(jid, index);
-	pb = strchr(parent_jid, (int)']');
-	strcat(jid, pb); /* "].hostname" section */
+	pb = strchr(jid, (int) '['); /* "seqnum[" section */
+	*(pb + 1) = '\0';
+	strncat(jid, index, sizeof(jid) - 1);
+	pb = strchr(parent_jid, (int) ']');
+	strncat(jid, pb, sizeof(jid) - 1); /* "].hostname" section */
```

Comment:
strncat() won't help here really.  It says it will copy at most N characters.  By saying sizeof(jid)-1 you could copy over the end of the array just the same.  The only way you're really going to save yourself is use snprintf().  That means you're going to have to put a null byte in the middle of parrent_jid and then put the character back.  I only brought this up as a possibility.  If you think this won't happen, you can go back to the previous code.